patches/minecraft/net/minecraft/item/ItemRedstone.java.patch

```diff
@@ -0,0 +1,11 @@
+--- ../src-base/minecraft/net/minecraft/item/ItemRedstone.java
++++ ../src-work/minecraft/net/minecraft/item/ItemRedstone.java
+@@ -25,7 +25,7 @@
+         BlockPos blockpos = flag ? p_180614_3_ : p_180614_3_.func_177972_a(p_180614_5_);
+         ItemStack itemstack = p_180614_1_.func_184586_b(p_180614_4_);
+ 
+-        if (p_180614_1_.func_175151_a(blockpos, p_180614_5_, itemstack) && p_180614_2_.func_190527_a(p_180614_2_.func_180495_p(blockpos).func_177230_c(), blockpos, false, p_180614_5_, (Entity)null) && Blocks.field_150488_af.func_176196_c(p_180614_2_, blockpos))
++        if (p_180614_1_.func_175151_a(blockpos, p_180614_5_, itemstack) && p_180614_2_.func_190527_a(p_180614_2_.func_180495_p(blockpos).func_177230_c(), blockpos, false, p_180614_5_, p_180614_1_) && Blocks.field_150488_af.func_176196_c(p_180614_2_, blockpos))
```

Comment:
> 
> 
> Are you 100% sure that changes like this do not impact vanilla behavior?

I have not seen any changes in vanilla behavior in the test I have done; However, I cannot say I have done every possible test for it myself. 

When I initially examined that segment of code when vanilla, it always pushed null values in that argument that were never used. I was very confused as to why it did that in the first place because of its lack of use. So to prevent any breaking changes, I decided to utilize the parameter to help fill in data for the event.

PS: Because it has been so long since this review, I may not remember the entire reasoning behind some of the changes. I am very busy and it will take some time to redo the edits to not have any merge conflicts as well.